python/pyspark/rdd.py

```diff
@@ -2483,6 +2485,32 @@ def _is_barrier(self):
         """
         return self._jrdd.rdd().isBarrier()
 
+    def withResources(self, profile):
+        """
+        .. note:: Experimental
+
+        Specify a ResourceProfile to use when calculating this RDD. This is only supported on
+        certain cluster managers and currently requires dynamic allocation to be enabled.
+        It will result in new executors with the resources specified being acquired to
+        calculate the RDD.
+
+        .. versionadded:: 3.0.0
```

Comment:
BTW, you want to backport this to `branch-3.0`?

Reply:
no, sorry I missed updating it, I'll fix it